### PR TITLE
add an email or phone transformer

### PR DESCRIPTION
**Pull request by danturn**
Heya y'all!

hope you're well, 

i had the need to be able to anonymise a field that was either a phone number OR an email (depending on a `type` variable in another column... i don't think we can couple one column to another so thought this was a better approach... 

wdyt? its pretty simple?

i also added all non General data types to the csv output as i think it makes more sense? you can always filter out the ones you dont need in the csv right?

lmk if this seems sensible!

Dan

### CODE REVIEW

**Comment by danturn** [comment]
@hoyon do you reckon y'all might look at this? if its not something you want  and dont want to merge it that is also fine!

File: README.md [hunk 1]
```diff
@@ -53,6 +53,8 @@ The following data categories are supported
 - FakeCity - Random city from [faker](https://github.com/cksac/fake-rs)
 - FakeCompanyName * - Random Company Name from [faker](https://github.com/cksac/fake-rs)
 - FakeEmail * - Random email address from [faker](https://github.com/cksac/fake-rs)
+- FakeEmailOrPhone * - Either a random phone number OR a random email depending on whether the existing data starts with a `+` and doesn't contain an `@` symbol or not!
+- Random email address from [faker](https://github.com/cksac/fake-rs)
```
Comment (hoyon) [inline README.md:57]:
```suggestion
- FakeEmailOrPhone * - Either a random phone number OR a random email depending on whether the existing data starts with a `+` and doesn't contain an `@` symbol or not!
```

**Review by hoyon — APPROVED** [review]

**Review by hoyon — CHANGES_REQUESTED** [review]
@danturn there are clippy issues. Can you take a look at fixing them? Otherwise it looks good.

**Comment by danturn** [comment]
> @danturn there are clippy issues. Can you take a look at fixing them? Otherwise it looks good.

@hoyon sorted, sorry thought i had done that already!

if you get a sec could you look at the gzip pr as well?

**Comment by danturn** [comment]
> > @danturn there are clippy issues. Can you take a look at fixing them? Otherwise it looks good.
> 
> @hoyon sorted, sorry thought i had done that already!
> 
> if you get a sec could you look at the gzip pr as well?

also, sorry @hoyon i cant see why the build is failing now, 

`build_and_test` runs without error for me locally..

**Comment by hoyon** [comment]
@danturn the build is failing due to a circleci permission issue. I'm not sure how to fix it. Let me merge another dependabot PR and then hit update branch and maybe it'll be happy with that.

**Review by hoyon — APPROVED** [review]

**Pull request merged**